tests/test_fulltextsearch.py

```diff
@@ -0,0 +1,108 @@
+"""Tests for full_text_search.py"""
+
+import unittest
+import logging
+import datetime
+
+from google.appengine.ext import db
+from google.appengine.ext import testbed
+from google.appengine.api import search
+
+import full_text_search
+import model
+
+TEST_DATETIME = datetime.datetime(2010, 1, 1, 0, 0, 0)
+
+def create_index(person):
+    full_text_search.create_index(
+        record_id=person.record_id,
+        repo=person.repo,
+        given_name=person.given_name,
+        family_name=person.family_name,
+        full_name=person.full_name,
+        alternate_names=person.alternate_names
+    )
+
+class FullTextSearchTests(unittest.TestCase):
+    def setUp(self):
+        self.tb = testbed.Testbed()
+        self.tb.activate()
+        self.tb.init_search_stub()
+        self.p1 = model.Person(
+            key_name='%s:%s' % ('haiti', 'haiti/0505'),
+            repo='haiti',
+            given_name='Iori',
+            family_name='Minase',
+            full_name='Iori Minase',
+            alternate_names='Iorin',
+            entry_date=TEST_DATETIME
+        )
+        self.p2 = model.Person(
+            key_name='%s:%s' % ('haiti', 'haiti/0325'),
+            repo='haiti',
+            given_name='Yayoi',
+            family_name='Takatsuki',
+            full_name='Yayoi Takatsuki',
+            alternate_names='Yayotan',
+            entry_date=TEST_DATETIME
+        )
+        self.p3 = model.Person(
+            key_name='%s:%s' % ('haiti', 'haiti/1202'),
+            repo='haiti',
+            given_name='Yayoi',
+            full_name='Yayoi san',
+            alternate_names='Nigochan',
+            entry_date=TEST_DATETIME
+        )
+        self.p4 = model.Person(
+            key_name='%s:%s' % ('haiti', 'haiti/1123'),
+            repo='haiti',
+            given_name='Miki',
+            family_name='Hoshii',
+            full_name='Miki Hoshii',
+            entry_date=TEST_DATETIME
+        )
+
+    def tearDown(self):
+        db.delete(self.p1)
+        self.tb.deactivate()
+
+    def test_create_index_name_only(self):
```

Comment:
The difference between test_add_record_to_index and test_search_by_name_only is still not clear to me. They both call add_record_to_index() and search(), right? Maybe you don't need test_add_record_to_index()?